### PR TITLE
config handling and Sam auth [risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
* Connect to Sam for first-pass auth.
* Preliminary features to support configuration.

This is a partial PR - it isn't fully functional until I get environment-specific deploy scripts working; right now it's hardcoded to talk to dev Sam. But I want  to keep  the PRs small and incremental.

### CODE REVIEW

File: function/dataaccess/sam.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const rp = require('request-promise-native');
+const {throwHttpError} = require('../errors');
+
+const checkUserEnabled = async (appConfig, authToken) => {
```
Comment (gpolumbo-broad) [inline function/dataaccess/sam.js:6]: Big method.  Consider refactoring into smaller methods.

Ah, yeah, I see your TODO further down now.

File: function/dataaccess/sam.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          }
+        } else {
+          throwHttpError(response.statusCode,
+              (response.body.message ? response.body.message : response.statusCode));
+        }
+      })
+      .catch((err) => {
+        // if we have an HttpError from a disabled user or non-200 from Sam, rethrow it
+        if (err.name === 'HttpError') {
+          throw err;
```
Comment (gpolumbo-broad) [inline function/dataaccess/sam.js:44]: https://cloud.google.com/functions/docs/monitoring/error-reporting 

Make sure you're throwing/recording errors per their guidelines

Is this going to run as Google Cloud Functions?  Or will it be deployed elsewhere?

**Review by gpolumbo-broad — CHANGES_REQUESTED** [review]
Sorry if it's a dumb question, but if this is going to be deployed as cloud functions, just make sure you're throwing and logging errors according to Google's guidelines.  Otherwise 👍

**Comment by davidangb** [comment]
thanks  @gpolumbo-broad !  Yes, this will be a cloud function (doc at https://docs.google.com/document/d/1DXvkL8NcmIDMAwN1_tB0RgQO6XJr0jf6xKiRXf2I3nQ).

I've updated the Express error handler to log errors. Everything should bubble up to there so it's a good central place to handle logging.

**Review by gpolumbo-broad — APPROVED** [review]

**Pull request merged**